public_html/wp-content/plugins/wordcamp-reports/classes/report/class-wordcamp-speaker-feedback.php

```diff
@@ -0,0 +1,393 @@
+		require_once $path . 'comment.php';
+		require_once $path . 'post.php';
+		require_once $path . 'view.php';
+		require_once $path . 'spam.php';
+		require_once $path . 'stats.php';
+	}
+
+	/**
+	 * Get data from a WordCamp Details report and filter it.
+	 *
+	 * @return array|null
+	 */
+	protected function get_wordcamps() {
+		$post_args = array(
+			'post_type'           => WCPT_POST_TYPE_ID,
+			'post_status'         => WordCamp_Loader::get_public_post_statuses(),
+			'posts_per_page'      => 9999,
+			'nopaging'            => true,
+			'no_found_rows'       => false,
+			'ignore_sticky_posts' => true,
+			'orderby'             => 'id',
+			'order'               => 'ASC',
+		);
+
+		if ( $this->range instanceof Date_Range ) {
+			// This replaces the default meta query.
+			$post_args['meta_query'] = array(
+				array(
+					'key'     => 'Start Date (YYYY-mm-dd)',
+					'value'   => array( $this->range->start->getTimestamp(), $this->range->end->getTimestamp() ),
+					'compare' => 'BETWEEN',
+					'type'    => 'NUMERIC',
+				),
+			);
+			$post_args['orderby'] = 'meta_value_num title';
+		}
+
+		if ( $this->wordcamp_id ) {
+			$post_args['post__in'] = $this->wordcamp_id;
+		}
+
+		return get_posts( $post_args );
+	}
+
+	/**
+	 * Render the page for this report in the WP Admin.
+	 *
+	 * @return void
+	 */
+	public static function render_admin_page() {
+		$start_date  = filter_input( INPUT_POST, 'start-date' );
+		$end_date    = filter_input( INPUT_POST, 'end-date' );
+		$wordcamp_id = filter_input( INPUT_POST, 'wordcamp-id' );
+		$refresh     = filter_input( INPUT_POST, 'refresh', FILTER_VALIDATE_BOOLEAN );
+
+		include get_views_dir_path() . 'report/wordcamp-speaker-feedback.php';
+	}
+
+	/**
+	 * Export the report data to a file.
+	 *
+	 * @return void
+	 */
+	public static function export_to_file() {
+		$start_date  = filter_input( INPUT_POST, 'start-date' );
+		$end_date    = filter_input( INPUT_POST, 'end-date' );
+		$wordcamp_id = filter_input( INPUT_POST, 'wordcamp-id' );
+		$refresh     = filter_input( INPUT_POST, 'refresh', FILTER_VALIDATE_BOOLEAN );
+		$action      = filter_input( INPUT_POST, 'action' );
+		$nonce       = filter_input( INPUT_POST, self::$slug . '-nonce' );
+
+		$report = null;
+
+		if ( 'Export CSV' !== $action ) {
+			return;
+		}
+
+		if ( wp_verify_nonce( $nonce, 'run-report' ) && current_user_can( 'manage_network' ) ) {
+			$options = array(
+				'public'         => false,
+				'earliest_start' => new DateTime( '2020-01-01' ), // Speaker Feedback Tool was introduced in 2020.
+				'max_interval'   => new DateInterval( 'P1Y1M' ),
+			);
+
+			if ( $refresh ) {
+				$options['flush_cache'] = true;
+			}
+
+			$report = new self( $start_date, $end_date, $wordcamp_id, $options );
+
+			$filename = array( $report::$name );
+			if ( $report->wordcamp_site_id ) {
+				$filename[] = get_wordcamp_name( $report->wordcamp_site_id );
+			}
+			$filename[] = $report->range->start->format( 'Y-m-d' );
```

Comment:
When selecting just a single WordCamp (i didn't pick a range), this is throwing an error:

> Fatal error: Uncaught Error: Call to a member function format() on null in /home/wordcamp/public_html/wp-content/plugins/wordcamp-reports/classes/report/class-wordcamp-speaker-feedback.php on line 347

Reply:
Oops. Unfortunately because of the way I originally designed these reports, the date range is required. Someday I'll do a refactor to make the report inputs more flexible. In the mean time, I added the `required` attribute to the inputs and handled the case there the date inputs are missing: ebec984a